lib/ClangImporter/ClangImporter.cpp

```diff
@@ -4005,6 +4006,578 @@ void ClangImporter::Implementation::lookupAllObjCMembers(
+                                 /*implicit*/ true);
+    casted->setType(InOutType::get(baseType));
+  } else {
+    auto *selfDecl = funcDecl->getImplicitSelfDecl();
+    auto selfExpr = new (ctx) DeclRefExpr(selfDecl, DeclNameLoc(),
+                                          /*implicit*/ true);
+    selfExpr->setType(selfDecl->getType());
+
+    auto staticCastRefExpr = getInteropStaticCastDeclRefExpr(
+        ctx, baseStruct->getClangDecl()->getOwningModule(), baseType,
+        derivedStruct->getDeclaredType());
+
+    auto castedCall = CallExpr::createImplicit(ctx, staticCastRefExpr,
+                                               {selfExpr}, {Identifier()});
+    castedCall->setType(baseType);
+    castedCall->setThrows(false);
+    casted = castedCall;
+  }
+
+  auto *baseMemberExpr =
+      new (ctx) DeclRefExpr(ConcreteDeclRef(baseMember), DeclNameLoc(),
+                            /*Implicit=*/true);
+  baseMemberExpr->setType(baseMember->getInterfaceType());
+
+  auto baseMemberDotCallExpr =
+      DotSyntaxCallExpr::create(ctx, baseMemberExpr, SourceLoc(), casted);
+  baseMemberDotCallExpr->setType(baseMember->getMethodInterfaceType());
+  baseMemberDotCallExpr->setThrows(false);
+
+  auto *baseMemberCallExpr = CallExpr::createImplicit(
+      ctx, baseMemberDotCallExpr, forwardingParams, {});
+  baseMemberCallExpr->setType(baseMember->getResultInterfaceType());
+  baseMemberCallExpr->setThrows(false);
+
+  auto returnStmt = new (ctx) ReturnStmt(SourceLoc(), baseMemberCallExpr,
+                                         /*implicit=*/true);
+
+  auto body = BraceStmt::create(ctx, SourceLoc(), {returnStmt}, SourceLoc(),
+                                /*implicit=*/true);
+  return {body, /*isTypeChecked=*/true};
+}
+
+// Getters are relatively easy. Just cast and return the member:
+//   %0 = __swift_interopStaticCast<Base>(self)
+//   return %0.member
+static std::pair<BraceStmt *, bool>
+synthesizeBaseClassFieldGetterBody(AbstractFunctionDecl *afd, void *context) {
+  ASTContext &ctx = afd->getASTContext();
+
+  AccessorDecl *getterDecl = cast<AccessorDecl>(afd);
+  VarDecl *baseClassVar = static_cast<VarDecl *>(context);
+  StructDecl *baseStruct =
+      cast<StructDecl>(baseClassVar->getDeclContext()->getAsDecl());
+  StructDecl *derivedStruct =
+      cast<StructDecl>(getterDecl->getDeclContext()->getAsDecl());
+
+  auto selfDecl = getterDecl->getImplicitSelfDecl();
+  auto selfExpr = new (ctx) DeclRefExpr(selfDecl, DeclNameLoc(),
+                                        /*implicit*/ true);
+  selfExpr->setType(selfDecl->getType());
+
+  auto staticCastRefExpr = getInteropStaticCastDeclRefExpr(
+      ctx, baseStruct->getClangDecl()->getOwningModule(),
+      baseStruct->getSelfInterfaceType(),
+      derivedStruct->getSelfInterfaceType());
+
+  auto casted = CallExpr::createImplicit(ctx, staticCastRefExpr, {selfExpr},
+                                         {Identifier()});
```

Comment:
Is this a new `CallExpr::createImplicit` API?

Reply:
Scratch that, looks like the API recently changed and this PR is using the old API.